app/handlers/testConfig.py

```
@@ -29,79 +29,121 @@ class TestConfigHandler(tornado.web.RequestHandler):
+
+class TestConfigObjectThread(threading.Thread):
+    def __init__(self, identifier, frame_start, num_frames, callback):
+        threading.Thread.__init__(self)
+        self.identifier = identifier
+        self.frame_start = frame_start
+        self.num_frames = num_frames
+        self.callback = callback
+
+    def run(self):
+        project_path = get_project_path(self.identifier)
         tracking_path = os.path.join(project_path, "tracking.cfg")
         obj_db_path = os.path.join(project_path,".temp", "test", "test_object", "test1.sqlite")
         feat_db_path = os.path.join(project_path, ".temp", "test", "test_feature", "test1.sqlite")
         if os.path.exists(obj_db_path):
             os.remove(obj_db_path)
         shutil.copyfile(feat_db_path, obj_db_path)
 
-        testing_dict = {'frame1': frame_start, 'nframes': num_frames}
+        testing_dict = {'frame1': self.frame_start, 'nframes': self.num_frames}
         update_config_without_sections(tracking_path, testing_dict)
 
-        images_folder = "object_images"
+        images_folder = os.path.join(get_project_path(self.identifier), "object_images")
         video.delete_files(images_folder)
 
         try:
             subprocess.call(["feature-based-tracking",tracking_path,"--gf","--database-filename",obj_db_path])
             subprocess.call(["classify-objects.py", "--cfg", tracking_path, "-d", obj_db_path])  # Classify road users
-            subprocess.call(["display-trajectories.py", "-i", get_project_video_path(identifier),"-d", obj_db_path, "-o", project_path + "/homography/homography.txt", "-t", "object", "--save-images", "-f", str(frame_start), "--last-frame", str(frame_start+num_frames)])
+            subprocess.call(["display-trajectories.py", "-i", get_project_video_path(self.identifier),"-d", obj_db_path, "-o", project_path + "/homography/homography.txt", "-t", "object", "--save-images", "-f", str(self.frame_start), "--last-frame", str(self.frame_start+self.num_frames)])
```

Unfortunately, try:except with subprocess.call will hide exceptions and return success. In this case we should instead use subprocess.check_output as this will raise errors that happen within the subprocess calls. @jabb1123 can fill you in on why we need this.

---

This change was already pushed to `dev`. I recommend pulling from `dev` to prevent conflicts or overwriting approved code.